opensrp-opd/src/main/java/org/smartregister/opd/activity/OpdProfileActivity.java

```diff
@@ -0,0 +1,304 @@
+package org.smartregister.opd.activity;
+
+import android.Manifest;
+import android.app.Activity;
+import android.content.Intent;
+import android.net.Uri;
+import android.support.annotation.NonNull;
+import android.support.v4.view.ViewPager;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import org.greenrobot.eventbus.EventBus;
+import org.smartregister.commonregistry.CommonPersonObjectClient;
+import org.smartregister.opd.R;
+import org.smartregister.opd.adapter.ViewPagerAdapter;
+import org.smartregister.opd.contract.OpdProfileActivityContract;
+import org.smartregister.opd.fragment.ProfileVisitFragment;
+import org.smartregister.opd.fragment.ProfileOverviewFragment;
+import org.smartregister.opd.presenter.OpdProfileActivityPresenter;
+import org.smartregister.opd.utils.OpdConstants;
+import org.smartregister.util.PermissionUtils;
+import org.smartregister.view.activity.BaseProfileActivity;
+
+import java.util.HashMap;
+
+/**
+ * Created by Ephraim Kigamba - ekigamba@ona.io on 2019-09-27
+ */
+
+public class OpdProfileActivity extends BaseProfileActivity implements OpdProfileActivityContract.View {
```

Comment:
Shouldn't  this be in the client app.....there are alot of commented out code here

Reply:
I am guessing there will not be any changes to the profile page. Since the table for the OPD is in the library, therefore this should be here will the option of customizing whatever might be different

Reply:
I thought that was the function of `BaseOpdProfileActivity`

Reply:
BTW, so what about I delete the `BaseOpdProfileActivity` since it is not shared and have `OpdProfileActivity` only. or the other way around. I am assuming any changes to the profile activity are configurable and added through builders or actions. Hopefully not by overriding the activity. What do you think?

Reply:
I was thinking that we can have a `BaseOpdProfileActivity` in opd and make it configurable, since there is only so much you can do with builders or actions...also using builders will make initialization tedious, or how do you see it?

Reply:
Cool. I merge the code into `BaseOpdProfileActivity`